src/core/server/csp/csp_config.test.mocks.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { config } from './config';
+
+const origSchema = config.schema;
+
+export const mockUnsafeEvalDefaultValue = Math.random() > 0.5 ? true : false;
+
+export const mockConfig = {
+  create() {
+    // @ts-expect-error: Property 'extends' does not exist on type??
+    config.schema = config.schema.extends({
+      unsafe_eval: schema.boolean({ defaultValue: mockUnsafeEvalDefaultValue }),
+    });
```

Comment:
I would have gone with `create(defaultUnsafeEval: boolean)`  and actively test both scenarios instead of using a random value in the test (which can be a source of flakiness)

Reply:
+1

Reply:
Much better idea. I'll do that 👍